packages/notifier/src/notifier.js

```diff
@@ -51,35 +51,26 @@ export const makeNotifier = sharableInternalsP => {
  * @returns {NotifierRecord<T>} the notifier and updater
  */
 export const makeNotifierKit = (...initialStateArr) => {
-  /** @type {PromiseRecord<UpdateRecord<T>>|undefined} */
-  let optNextPromiseKit;
-  /** @type {UpdateCount} */
-  let currentUpdateCount = 1; // avoid falsy numbers
-  /** @type {UpdateRecord<T>|undefined} */
-  let currentResponse;
-
-  const hasState = () => currentResponse !== undefined;
-
-  const final = () => currentUpdateCount === undefined;
+  const { publisher, subscriber } = makeEmptyPublishKit();
 
+  /**
+   * @template T
+   * @type {BaseNotifier<T>}
+   */
   const baseNotifier = Far('baseNotifier', {
-    // NaN matches nothing
-    getUpdateSince(updateCount = NaN) {
-      if (
-        hasState() &&
-        (final() ||
-          (currentResponse && currentResponse.updateCount !== updateCount))
-      ) {
-        // If hasState() and either it is final() or it is
-        // not the state of updateCount, return the current state.
-        assert(currentResponse !== undefined);
-        return Promise.resolve(currentResponse);
-      }
-      // otherwise return a promise for the next state.
-      if (!optNextPromiseKit) {
-        optNextPromiseKit = makePromiseKit();
+    getUpdateSince(sinceUpdateCount = undefined) {
```

Comment:
I can't think of any good reason to change the API for `getUpdateSince`.
```suggestion
    getUpdateSince(sinceUpdateCount = NaN) {
```

Reply:
Per https://github.com/Agoric/agoric-sdk/pull/5435#discussion_r898229051 , "_the contract is that the argument to getUpdateSince() can be something that was previously published as updateCount. If it's anything but the current count, it's ignored_.". Is there still code violating this by explicitly sending `NaN` or another value not extracted from `updateCount`, or expecting `updateCount` to be a number?